include/drivers/dai.h

```diff
@@ -0,0 +1,629 @@
+ * @retval -ENOSYS DAI_DIR_BOTH value is not supported.
+ */
+static inline int dai_z_trigger(const struct device *dev,
+				enum dai_dir dir,
+				int stream_id,
+				enum dai_trigger_cmd cmd)
+{
+	const struct dai_driver_api *api =
+		(const struct dai_driver_api *)dev->api;
+
+	return api->trigger(dev, dir, stream_id, cmd);
+}
+
+/**
+ * @brief Read data from the RX queue into a provided buffer
+ *
+ * Data received by the DAI interface is stored in the RX queue consisting of
+ * memory blocks preallocated in init. Calling this function removes one block
+ * from the queue which is copied into the provided buffer and then freed.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param buf Destination buffer for read data, which must be at least
+ *            as large as the configured memory block size for the RX channel.
+ * @param size Pointer to the variable storing the number of bytes read.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO The interface is in NOT_READY or ERROR state and there are no
+ *         more data blocks in the RX queue.
+ * @retval -EBUSY Returned without waiting.
+ * @retval -EAGAIN Waiting period timed out.
+ */
+static inline int dai_z_read(const struct device *dev, void *buf, size_t *size)
+{
+	const struct dai_driver_api *api =
+		(const struct dai_driver_api *)dev->api;
+
+	return api->read(dev, buf, size);
+}
+
+/**
+ * @brief Write data to the TX queue from a provided buffer
+ *
+ * This function acquires a memory block from the DAI TX queue
+ * and copies the provided data buffer into it.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param buf Pointer to a buffer containing the data to transmit.
+ * @param size Number of bytes to write. This value has to be equal or smaller
+ *        than the size of the channel's TX memory block configuration.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO The interface is not in READY or RUNNING state.
+ * @retval -EBUSY Returned without waiting.
+ * @retval -EAGAIN Waiting period timed out.
+ * @retval -ENOMEM No memory in TX slab queue.
+ * @retval -EINVAL Size parameter larger than TX queue memory block.
+ */
+static inline int dai_z_write(const struct device *dev, void *buf, size_t size)
+{
+	const struct dai_driver_api *api =
+		(const struct dai_driver_api *)dev->api;
+
+	return api->write(dev, buf, size);
+}
+
+static inline int dai_z_ts_config(const struct device *dev, struct dai_ts_cfg *cfg)
+{
+	const struct dai_driver_api *api =
+		(const struct dai_driver_api *)dev->api;
+
+	if (!api->ts_config)
+		return -EINVAL;
+
+	return api->ts_config(dev, cfg);
+}
+
+static inline int dai_z_ts_start(const struct device *dev, struct dai_ts_cfg *cfg)
+{
+	const struct dai_driver_api *api =
+		(const struct dai_driver_api *)dev->api;
+
+	if (!api->ts_start)
+		return -EINVAL;
+
+	return api->ts_start(dev, cfg);
+}
+
+static inline int dai_ts_z_stop(const struct device *dev, struct dai_ts_cfg *cfg)
```

Comment:
Was `dai_z_ts...` previously, but switched to `dai_ts_z_...()` (this function and the next one) - why?

Reply:
bug.. need to fix